integration/system/stats_test.go

```diff
@@ -161,32 +124,57 @@ func (s *DockerAPISuite) TestAPIStatsNetworkStats(c *testing.T) {
 		expRxPkts++
 		expTxPkts++
 	}
-	assert.Assert(c, postTxPackets >= expTxPkts, "Reported less TxPackets than expected. Expected >= %d. Found %d. %s", expTxPkts, postTxPackets, pingouts)
-	assert.Assert(c, postRxPackets >= expRxPkts, "Reported less RxPackets than expected. Expected >= %d. Found %d. %s", expRxPkts, postRxPackets, pingouts)
+	assert.Assert(t, postTxPackets >= expTxPkts, "Reported less TxPackets than expected. Expected >= %d. Found %d. %s", expTxPkts, postTxPackets, pingouts)
+	assert.Assert(t, postRxPackets >= expRxPkts, "Reported less RxPackets than expected. Expected >= %d. Found %d. %s", expRxPkts, postRxPackets, pingouts)
 }
 
-func (s *DockerAPISuite) TestAPIStatsNetworkStatsVersioning(c *testing.T) {
-	// Windows doesn't support API versions less than 1.25, so no point testing 1.17 .. 1.21
-	testRequires(c, testEnv.IsLocalDaemon, DaemonIsLinux)
+func TestAPIStatsNoStreamConnectedContainers(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
 
-	id := runSleepingContainer(c)
-	cli.WaitRun(c, id)
-	wg := sync.WaitGroup{}
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
 
-	for i := 17; i <= 21; i++ {
-		wg.Add(1)
-		go func(i int) {
-			defer wg.Done()
-			apiVersion := fmt.Sprintf("v1.%d", i)
-			statsJSONBlob := getVersionedStats(c, id, apiVersion)
-			if versions.LessThan(apiVersion, "v1.21") {
-				assert.Assert(c, jsonBlobHasLTv121NetworkStats(statsJSONBlob), "Stats JSON blob from API %s %#v does not look like a <v1.21 API stats structure", apiVersion, statsJSONBlob)
-			} else {
-				assert.Assert(c, jsonBlobHasGTE121NetworkStats(statsJSONBlob), "Stats JSON blob from API %s %#v does not look like a >=v1.21 API stats structure", apiVersion, statsJSONBlob)
-			}
-		}(i)
+	cID1 := container.Run(ctx, t, apiClient, container.WithImage("busybox"), container.WithCmd("sleep", "20"))
+	cID2 := container.Run(ctx, t, apiClient, container.WithImage("busybox"), container.WithCmd("top"), container.WithNetworkMode("container:"+cID1))
+
+	ch := make(chan error, 1)
+	go func() {
+		resp, body, err := request.Get(testutil.GetContext(t), "/containers/"+cID2+"/stats?stream=false")
+		defer body.Close()
+		if err != nil {
+			ch <- err
```

Comment:
Seems like all these err sends need a return statement.
Maybe something like this would be more ergonomic:

```go
go func() {
    ch <- func() error {
        // logic with err return
    }
}()
```